`Development`: Add documentation and feature overview for test exams and exam import (#5610)
ls1intum/Artemis: docs/user/exams/instructors_guide.rst

- You can create an exam by clicking on |create_new_exam|.

> 1.2 Configure Exam

Reviewer: There are three references in this file that link to the old section name by name
- line 25
- line 471
- line 500

Those references should be updated to use a proper `:ref:` to a defined id.
```suggestion
_configure_exam:

1.2 Configure Exam
```

The ref name for 1.2.1 Create Exam should then be renamed to avoid conflicting ids.

^^^^^^^^^^^^^^^^^^^

> **1.2.1 Exam Mode**

Reviewer: ```suggestion

1.2.1 Exam Mode
"""""""""""""""
```

The section should be properly marked as such. Same for a few of the following ones.

- Artemis supports two exam modes. The conduction of normal exams and test exams.

> - The normal exam mode is suitable for conducting end-of-semester exams. Students can view and edit the exam between the configured working time. Afterwards you can perform a manual or automated evaluation of the students' submissions. The results will be published on the specified date.

Reviewer: ```suggestion
- The normal exam mode is suitable for conducting end-of-semester exams. Students can view and work on the exam between the configured working time. Afterwards you can perform a manual or automated evaluation of the students' submissions. The results will be published on the specified date.
```

> - Test exams provide you with the possibility to provide students with a practice opportunity for the end-of-semester exam. The main difference is that you choose an working window within which students can freely start the exam. Students then have the configured working time to complete the exam. After submitting, students immediately get the automated assessment for programming and quizzes. Manual correction of the submissions is not necessary.

Reviewer: ```suggestion
- Test exams provide you with the possibility to provide students with a practice opportunity for the end-of-semester exam. The main difference is that you choose a working window within which students can freely start the exam. Students then have the configured working time to complete the exam. After submitting, students immediately get the automated assessment for programming and quizzes. Manual correction of the submissions is not necessary.
```

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1.2.1 Exam Mode

> ^^^^^^^^^^^^^^^

Reviewer: ```suggestion
"""""""""""""""
```

- Artemis supports two exam modes. The conduction of normal exams and test exams.
- The normal exam mode is suitable for conducting end-of-semester exams. Students can view and work on the exam between the configured working time. Afterwards you can perform a manual or automated evaluation of the students' submissions. The results will be published on the specified date.
- Test exams provide you with the possibility to provide students with a practice opportunity for the end-of-semester exam. The main difference is that you choose a working window within which students can freely start the exam. Students then have the configured working time to complete the exam. After submitting, students immediately get the automated assessment for programming and quizzes. Manual correction of the submissions is not necessary.

.. _create_exam:

> 1.2.2 Create Exam
> ^^^^^^^^^^^^^^^^^^

Reviewer: ```suggestion
1.2.2 Create Exam
"""""""""""""""""
```

   Exam Assessment and Student Review

**1.2.3 Import Exam**

- Alternatively, you can import an existing exam by clicking on |import_exam|.
- Artemis displays a list of all available exams. To select one specific exam for the import, click on the |import_individual_exam| button.

.. figure:: instructor/exam_import_selection.png
   :alt: List with all exams available for import
   :align: center

> - You are now presented with the *Import Exam* view. All information except for the dates are copied from the exam, you selected for the import. You can find more informations regarding this view at the section :ref:`create and configure an exam <_configure_exam>`.

Reviewer: ```suggestion
- You are now presented with the *Import Exam* view. All information except for the dates are copied from the exam, you selected for the import. You can find more informations regarding this view at the section :ref:`create and configure an exam <configure_exam>`.
```
The reference does not need the underscore here but only at the definition side.

> - You are now presented with the *Import Exam* view. All information except for the dates are copied from the exam, you selected for the import. You can find more informations regarding this view at the section :ref:`create and configure an exam <_configure_exam>`.

Reviewer: ```suggestion
- You are now presented with the *Import Exam* view. All information except for the dates are copied from the exam you selected for the import. You can find more information regarding this view at the section :ref:`create and configure an exam <_configure_exam>`.
```
2x typos

> - Additionally, you can select or deselect exercises, which are imported alongside the exam.

Reviewer: ```suggestion
- Additionally, you can select or deselect exercises which are imported alongside the exam.
```

> - You can change the `title` and `isMandatory` of an exercise group, as well as the `title` (and `short-name` for programming exercises) for the individual exercises.

Reviewer: ```suggestion
- You can change the ``title`` and ``isMandatory`` of an exercise group, as well as the ``title`` (and ``short-name`` for programming exercises) for the individual exercises.
```
*Italics* vs ``monospace``? In 1.2.2 monospace is used. It should be made consistent in one style.

- The title and short name of programming exercises must be unique. If you want to import an exercise group into the same course, you must change the title and short name before you can import the exercise group.
- After you have started the import by clicking on |import_exam|, Artemis checks if the title and short name of the selected programming exercise(s) are unique. If they are not unique, a warning is displayed and you have to change the corresponding title and short name.
.. note::
    - Further changes to the individual exercises can be made after the import by editing the respective exercise.

>     - Programming exercises are imported using their initial configuration. This import functionality cannot be used for changing the submission policy, for activating / deactivating the static code analysis or for creating new build plans. In this case, please import the exercises individually into the exercise groups.

Reviewer: ```suggestion
    - Programming exercises are imported using their initial configuration. This import functionality cannot be used for changing the submission policy, for activating/deactivating the static code analysis, or for creating new build plans. In this case, please import the exercises individually into the exercise groups.
```

   Exam Assessment and Student Review

**1.2.3 Import Exam**
^^^^^^^^^^^^^^^^^^^^^^

> - Alternatively, you can import an existing exam by clicking on |import_exam|.

Reviewer: The section starts with ‘Alternatively, […]’. At that point it is not obvious as alternative to what.

```suggestion
- Instead of creating a new exam, you can import an existing exam by clicking on |import_exam| from any of the courses you are an instructor/editor (?) in.
```

- Artemis displays a list of all available exams. To select one specific exam for the import, click on the |import_individual_exam| button.

.. figure:: instructor/exam_import_selection.png
   :alt: List with all exams available for import
   :align: center

- You are now presented with the *Import Exam* view. All information except for the dates are copied from the exam you selected for the import. You can find more information regarding this view at the section :ref:`create exam <create_exam>`.

> - Additionally, you can select or deselect exercises which are imported alongside the exam. You can find more informations regarding the exercise import in :ref:`section regarding the exercise group import <exercise_import>`.

Reviewer: ```suggestion
- Additionally, you can select or deselect exercises which are imported alongside the exam. You can find more information regarding the exercise import in the :ref:`section regarding the exercise group import <exercise_import>`.
```

1.2.3 Import Exam

> ^^^^^^^^^^^^^^^^^

Reviewer: ```suggestion
"""""""""""""""""
```

   Exercise Groups with different Exercise Variants

> 1.3.1 Import Exercise Groups
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Reviewer: ```suggestion
1.3.1 Import Exercise Groups
""""""""""""""""""""""""""""
```

- Artemis exam mode allows you to import one or more exercise groups from an existing exam.
- The import process consists of two steps.

**Step 1: Select Exam**

- When you click on |import_exercise_group|, you can select one exam from which exercise group(s) should be imported.
- To select one exam, click on |select_exercise_group|.

.. figure:: instructor/exercise_group_import_exam.png
   :alt: First step in the process to import exercise group(s). You can select one exam, from which exercise group(s) should be imported
   :align: center

.. _exercise_import:

**Step 2: Select Exercises and Exercise Groups**

> 

Reviewer: ```suggestion
```
Nitpick: two empty lines.

- In the next step you can select or deselect exercises which should be imported alongside the exercise groups.
- You can also change the ``title`` and ``isMandatory`` of an exercise group, as well as the ``title`` (and ``short-name`` for programming exercises) for the individual exercises.
- The title and short name of programming exercises must be unique. If you want to import an exercise group into the same course, you must change the title and short name before you can import the exercise group.
- After you have started the import by clicking on |import_exam|, Artemis checks if the title and short name of the selected programming exercise(s) are unique. If they are not unique, a warning is displayed and you have to change the corresponding title and short name.

> .. note::

Reviewer: ```suggestion

.. note::
```
I get a warning when building the docs:
```
instructors_guide.rst:148: WARNING: Bullet list ends without a blank line; unexpected unindent.
```

   Exercise Groups with different Exercise Variants

**1.3.1 Import Exercise Groups**

- Artemis exam mode allows you to import one or more exercise groups from an existing exam.

> - The import process consists of a two-step process.

Reviewer: ```suggestion
- The import process consists of two steps.
```

> - When you click on |import_exercise_group|, you can select one exam, from which exercise group(s) should be imported.

Reviewer: ```suggestion
- When you click on |import_exercise_group|, you can select one exam from which exercise group(s) should be imported.
```

- To select one exam, click on |select_exercise_group|.

.. figure:: instructor/exercise_group_import_exam.png
   :alt: First step in the process to import exercise group(s). You can select one exam, from which exercise group(s) should be imported
   :align: center

> - In the next step, you can select or deselect exercises, which should be imported alongside the exercise groups.

Reviewer: ```suggestion
- In the next step you can select or deselect exercises which should be imported alongside the exercise groups.
```

> - You can also change the `title` and `isMandatory` of an exercise group, as well as the `title` (and `short-name` for programming exercises) for the individual exercises.

Reviewer: Italics vs monospace here as well.

- The title and short name of programming exercises must be unique. If you want to import an exercise group into the same course, you must change the title and short name before you can import the exercise group.
- After you have started the import by clicking on |import_exam|, Artemis checks if the title and short name of the selected programming exercise(s) are unique. If they are not unique, a warning is displayed and you have to change the corresponding title and short name.
.. note::
    - Further changes to the individual exercises can be made after the import by editing the respective exercise.

>     - Programming exercises are imported using their initial configuration. This import functionality cannot be used for changing the submission policy, for activating / deactivating the static code analysis or for creating new build plans. In this case, please import the exercises individually into the exercise groups.

Reviewer: 1:1 Repetition from section 1.2.3 above.

5. **Student Review**
---------------------

> During the review period, students have the opportunity to review the assessment of their exam. If they find inconsistencies, they can submit complaints about perceived mistakes made in the :ref:`assessment <exam_assessment>`. Students can provide their reasoning through a text message to clarify their objections. You can set the student review period in the :ref:`create exam <create_exam>` during the :ref:`exam configuration <exam_creation_and_configuration>`.

Reviewer: ```suggestion
During the review period, students have the opportunity to review the assessment of their exam. If they find inconsistencies, they can submit complaints about perceived mistakes made in the :ref:`assessment <exam_assessment>`. Students can provide their reasoning through a text message to clarify their objections. You can set the student review period when :ref:`creating an exam <create_exam>` during the :ref:`exam configuration <exam_creation_and_configuration>`.
```
